File: src/test/java/com/appland/appmap/AppmapPluginTest.java
package com.appland.appmap;

import org.gradle.api.Project;
import org.gradle.testfixtures.ProjectBuilder;
import org.junit.jupiter.api.Test;


public class AppmapPluginTest {
  @Test
  public void test() {
    // Given
    Project project = ProjectBuilder.builder().build();

    // When
    project.getPlugins().apply("com.appland.appmap");

  }
}

File: src/test/kotlin/com/appland/appmap/AppmapPluginTest.kt
package com.appland.appmap

import org.gradle.api.Project
import org.junit.Assert.*
import org.junit.Test
import org.gradle.testfixtures.*

class AppmapPluginTest {

    @Test
    fun apply() {
        val project: Project = ProjectBuilder.builder().build()
        project.pluginManager.apply("com.appland.appmap-plugin")
        assertTrue(project.pluginManager.hasPlugin("com.appland.appmap-plugin"))
        assertNotNull(project.tasks.getByName("appmap"))
    }
}